tests/infer/mcmc/test_mh.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import absolute_import, division, print_function
+
+import pytest
+import torch
+from torch.autograd import Variable
+
+import pyro
+import pyro.infer
+from pyro.distributions import Normal
+from tests.common import TestCase
+from pyro.infer.abstract_infer import Marginal
+from pyro.infer.mcmc.mh import MH, NormalProposal
+from pyro.infer.mcmc.mcmc import MCMC
+from pyro.util import ng_ones, ng_zeros
+
+
+class NormalNormalSamplingTestCase(TestCase):
+
+    def setUp(self):
+
+        pyro.clear_param_store()
+
+        def model():
+            mu = pyro.sample("mu", Normal(Variable(torch.zeros(1)),
+                                          Variable(torch.ones(1))))
+            xd = Normal(mu, Variable(torch.ones(1)), batch_size=50)
+            pyro.observe("xs", xd, self.data)
+            return mu
+
+        # data
+        self.data = Variable(torch.zeros(50, 1))
```

Comment:
nit: I would recommend a dataset for which it is more difficult to accidentally guess the parameters (ie with buggy code). For example, put 2 points on +1 and 1 point on +2. Then the posterior mean is 1.0 and stddev is sqrt(0.5). (I just worry that with the current values, 1/50 and 1/51 are indistinguishable, so you can't tell whether the prior has been included)

Reply:
As this line comes from [an existing test](https://github.com/uber/pyro/blob/dev/tests/infer/test_sampling.py#L59), can you explain the intuitions for why it's okay for an importance sampling test, but isn't very good for MH? I don't think I understand the subtlety, and it might help me to write improved MH tests.

Reply:
The intuition here is just that the importance sampling test (which I wrote in a hurry a long time ago) isn't a particularly good test.